### PR TITLE
Adding integration tests for response modes including JARM response modes

### PR DESCRIPTION
**Pull request by anjuchamantha**
This PR contains the integration tests to test response modes in /authorize call. Tests include both plain response mode tests as well as JARM response mode tests

Issue: https://github.com/wso2/product-is/issues/16724

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    /**
+     * Extract authorization code.
+     * @param url redirection url
+     * @param responseMode response mode which is used to send the request
+     * @return AuthorizationCode object
+     */
+    private AuthorizationCode getAuthorizationCode(String url, String responseMode)
+            throws URISyntaxException {
+        String code;
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java:186]:
```suggestion

        String code;
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        if (responseMode.contains("jwt")) {
+            String responseJWT;
+            Assert.assertTrue(url.contains("response"),
+                    "Response JWT not found in the response.");
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java:190]:
```suggestion
            Assert.assertTrue(url.contains("response"), Response JWT not found in the response.");
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.ApplicationResponseModel;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.InboundProtocols;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.OpenIDConnectConfiguration;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.util.*;
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java:53]: let's not use star imports

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java (hunk 1 above)
Comment (anjuchamantha) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java:190]: fixed with [52a5410](https://github.com/wso2/product-is/pull/16726/commits/52a5410ae119993b886fa486c661dfc5c7cfbe8f)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java (hunk 2 above)
Comment (anjuchamantha) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java]: fixed with [52a5410](https://github.com/wso2/product-is/pull/16726/commits/52a5410ae119993b886fa486c661dfc5c7cfbe8f)

**Review by chamathns — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java (hunk 3 above)
Comment (anjuchamantha) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ResponseModeTestCase.java]: fixed with [52a5410](https://github.com/wso2/product-is/pull/16726/commits/52a5410ae119993b886fa486c661dfc5c7cfbe8f)

**Pull request merged**